### PR TITLE
Stage level validation - Joiner widgets

### PR DESCRIPTION
**Pull request by itsanudeep**
This PR is to make changes to joiner widgets for stage level validation integration.

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/FieldRow.tsx [hunk 1]
```diff
@@ -74,30 +96,40 @@ const FieldRow: React.FC<IFieldRowProps> = ({
...
+              fullWidth
+              className={classes.textFieldCell}
+              margin="dense"
+              variant="outlined"
+              value={field.alias}
+              onChange={aliasChange}
+              disabled={!field.selected || disabled}
+            />
+          </div>
+          <div className={classes.errorText}>{validationError}</div>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/FieldRow.tsx]: wrap this with `If`?

Comment (itsanudeep) [reply cdap-ui/app/cdap/components/AbstractWidget/SqlSelectorWidget/FieldRow.tsx]: done!

**Comment by elfenheart** [comment]
@itsanudeep you still have to fix this branch to not branch off from develop, but release.

**Comment by itsanudeep** [comment]
@elfenheart rebased with release/6.1

**Review by elfenheart — APPROVED** [review]
👍 LGTM

**Review by yukiej — APPROVED** [review]
lgtm

File: cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx [hunk 2]
```diff
@@ -112,50 +119,55 @@ const Rule: React.FC<IRuleProps> = ({
...
+            <span className={classes.andBlock}>AND</span>
+          </If>
+          <If condition={last}>
+            <IconButton disabled={disabled} size="small" onClick={addRule}>
+              <AddIcon fontSize="large" />
+            </IconButton>
+          </If>
+        </div>
       </div>
+      <If condition={error}>
```
Comment (yukiej) [inline cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx]: I'm seeing a TS error here, about the types of condition needing to be boolean (and it is currently being assigned to string?)

Comment (itsanudeep) [reply cdap-ui/app/cdap/components/AbstractWidget/SqlConditionsWidget/Rule.tsx]: fixed!

**Comment by itsanudeep** [comment]
RAT and Checkstyle: https://builds.cask.co/browse/CDAP-DRC9544-5
Build and Test: https://builds.cask.co/browse/CDAP-UDUT417

**Pull request merged**